runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -350,8 +351,20 @@ private Options getPersistentMapOptions() {
     Table<K, V, M> getTable(String namespace, String tableName) {
         String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
         if (!tableMap.containsKey(fullyQualifiedTableName)) {
-            throw new NoSuchElementException(
-                    String.format("No such table found: namespace: %s, tableName: %s", namespace, tableName));
+            // Table has not been opened, but let's first find out if this table even exists
+            // in the corfu cluster
+            UUID tableStreamId = CorfuRuntime.getStreamID(fullyQualifiedTableName);
+            long tableStreamTail = runtime.getSequencerView().query(tableStreamId);
+            if (tableStreamTail != Address.NON_EXIST) {
```

Comment:
The checkpointer can bump the tail even if the table is empty, so not sure if this check is reliable.

Reply:
I think the consumer might be more interested in the life cycle of the table, so an explicit register operation from the application and not the checkpointer (since that's just a consequence of how checkpointing is done today, but it really shouldnt be that way).

Reply:
The goal is to just distinguish between a table that was never created from the case where a table was created. Empty table is also a valid table. Checkpointer cannot checkpoint a table that has never been created and that's all the caller needs to worry about for this change.

Reply:
Lifecycle management is outside the scope of this bug fix.

Reply:
Whether you want to implement the lifecycle of the stream or not, this patch doesn't seem to solve the issue that you are trying to fix. The application itself cannot differentiate between whether a table was created by the application, or the checkpointer. Before the application opens the table, the checkpointer can write an empty checkpoint for that stream and `tableStreamTail != Address.NON_EXIST` will always be false.

Reply:
If the checkpointer is able to see a table it means that the table has been created in which case this api would correctly return that the table exists. Note that the caller is only interested in checking if a table exists not if it is empty. So why won't this patch work?